backend/benefit/applications/api/v1/ahjo_integration_views.py

```diff
@@ -0,0 +1,140 @@
+from applications.enums import AhjoStatus as AhjoStatusEnum
+from applications.models import AhjoStatus, Application, Attachment
+from common.permissions import SafeListPermission
+from shared.audit_log import audit_logging
+from shared.audit_log.enums import Operation
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AhjoAttachmentView(APIView):
+    authentication_classes = [
+        TokenAuthentication,
+    ]
+    permission_classes = [IsAuthenticated, SafeListPermission]
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="uuid",
+                description="UUID of the attachment",
+                required=True,
+                type=OpenApiTypes.UUID,
+                location=OpenApiParameter.PATH,
+            )
+        ],
+        description="Returns the specified attachment file if it exists.",
+    )
+    def get(self, request, *args, **kwargs):
+        attachment_id = self.kwargs["uuid"]
+
+        try:
+            attachment = Attachment.objects.get(id=attachment_id)
+            audit_logging.log(
+                request.user,
+                "",  # Optional user backend
+                Operation.READ,
+                attachment,
+                additional_information="attachment was sent to AHJO!",
+            )
+            return self._prepare_file_response(attachment)
+        except Attachment.DoesNotExist:
+            return Response(
+                {"message": "Attachment not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+    @staticmethod
+    def _prepare_file_response(attachment: Attachment) -> FileResponse:
+        file_handle = attachment.attachment_file.open()
+        response = FileResponse(file_handle, content_type=f"{attachment.content_type}")
+        response["Content-Length"] = attachment.attachment_file.size
+        response[
+            "Content-Disposition"
+        ] = f"attachment; filename={attachment.attachment_file.name}"
+        return response
+
+
+class AhjoCallbackView(APIView):
+    authentication_classes = [
+        TokenAuthentication,
+    ]
+    permission_classes = [IsAuthenticated, SafeListPermission]
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="uuid",
+                description="UUID of the application",
+                required=True,
+                type=OpenApiTypes.UUID,
+                location=OpenApiParameter.PATH,
+            )
+        ],
+        description="Callback endpoint for Ahjo to send updates to the application.",
+        request=AhjoCallbackSerializer,
+        responses={
+            (status.HTTP_200_OK, "text/json"): OpenApiTypes.OBJECT,
+            (status.HTTP_404_NOT_FOUND, "text/json"): OpenApiTypes.OBJECT,
+        },
+    )
+    def post(self, request, *args, **kwargs):
+        serializer = AhjoCallbackSerializer(data=request.data)
+
+        if serializer.is_valid():
```

Comment:
I think this would be much more readable if it would be like:
```
if not serializer.is_valid():
    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)

do stuff if valid
```